packages/paste-core/components/data-grid/stories/components/PaginatedDataGrid.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from 'react';
+import {Box} from '@twilio-paste/box';
+import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {CheckboxGroup, Checkbox} from '@twilio-paste/checkbox';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {
+  Pagination,
+  PaginationItems,
+  PaginationArrow,
+  PaginationNumbers,
+  PaginationNumber,
+  PaginationEllipsis,
+} from '@twilio-paste/pagination';
+import {DataGrid, DataGridHead, DataGridRow, DataGridHeader, DataGridBody, DataGridCell} from '../../src';
+import {TableHeaderData, PaginatedTableBodyData} from './constants';
+
+const getRange = (start: number, end: number): number[] => {
+  return [...new Array(end - start + 1)].map((_, index) => index + start);
+};
+
+/* Calculates the correct display of the pagination numbers */
+/* TODO: move this into the Pagination package as an export with tests */
```

Comment:
You thinking like a hook potentially or something like that? It's pretty neat

Reply:
Haven't decided on the form yet, but _something_ that lets people use Pagination the way we intended. This was complicated code to write, would be hard for teams to get it right consistently